app_api/tests/__init__.py

```diff
@@ -61,6 +61,10 @@ def setUp(self):  # noqa
         self.trans = connection.begin()
 
         # bind an individual Session to the connection
+        # Next line is needed to make several tests run in a row.
+        # See https://github.com/Pylons/webtest/issues/5
+        # FIXME Is there a better solution?
+        DBSession.remove()
```

Comment:
That part is weird. But that entire setup process looks weird to me. I'd like to take a look at it, but I suggest that you merge your branch first.